core/src/main/scala/org/apache/spark/resource/ResourceUtils.scala

```diff
@@ -392,11 +392,13 @@ private[spark] object ResourceUtils extends Logging {
       s"${resourceRequest.id.resourceName}")
   }
 
-  def validateTaskCpusLargeEnough(execCores: Int, taskCpus: Int): Boolean = {
+  def validateTaskCpusLargeEnough(sparkConf: SparkConf, execCores: Int, taskCpus: Int): Boolean = {
     // Number of cores per executor must meet at least one task requirement.
-    if (execCores < taskCpus) {
-      throw new SparkException(s"The number of cores per executor (=$execCores) has to be >= " +
-        s"the number of cpus per task = $taskCpus.")
+    if (!sparkConf.get(TASKSET_MANAGER_SPECULATION_TESTING)) {
```

Comment:
so this conf if for testing speculation to specifically test the case when we don't know what the executor cores setting is - in like standalone mode:     

 skip throwing exception when cores per task > cores per executor to emulate standalone mode

I did this to emulate the standalone mode behavior even though running in local mode since we don't really have tests for full standalone mode. In standalone mode this function wouldn't even be called because we skip it. 

Let me see if there is a cleaner way to test this as to not cause confusion.

Reply:
Found a bit cleaner way that isn't called as much and localized to running in local mode. Added comments to the config so hopefully its clearer.  We only want to skip for unit testing purposes